keyboards/skyloong/gk87/bl/bl.c

```diff
@@ -0,0 +1,82 @@
+//  Copyright 2023 NaturalZh (@NaturalZh)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include "quantum.h"
+
+bool DIS_BRETH = 0;
+
+void suspend_power_down_kb() {
+    writePinHigh(MAC_PIN);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb() {
+    suspend_wakeup_init_user();
+}
+
+bool shutdown_kb(bool jump_to_bootloader) {
+    writePinHigh(MAC_PIN);
+    return shutdown_user(jump_to_bootloader);
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+       case DF(0):
+            set_single_persistent_default_layer(0); // Save default layer 0 to eeprom
+         return true; // continue all further processing of this key
```

Comment:
Should return false, as otherwise, it's processing the layer keycode twice.

Also, should check for `record->event.pressed`, otherwise, it triggers on press and release.

Reply:
OK, I corrected it. Thank you.